File: Domain/Product.cs
using System;
using System.Collections.Generic;

namespace Domain
{
  public class Product
  {
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Image { get; set; }
    public string Description { get; set; }
    public string Category { get; set; }
    public string City { get; set; }
    public string State { get; set; }
    public float Price { get; set; }
    public virtual ICollection<UserProduct> UserProducts { get; set; }
    public virtual ICollection<Photo> ProductPhotos { get; set; } //1:m relationship with Photos

  }
}

File: client/src/app/models/product.ts
export interface IProduct {
  id: string;
  title: string;
  image: string;
  description: string;
  category: string;
  city: string;
  state: string;
  price: number | null;
  isLiked: boolean;
  isChef: boolean;
  likes: ILiker[];
  photos: IPhoto[];
}

export class ProductFormValues implements Partial<IProduct> {
  id: string = "";
  title: string = "";
  image: string = "";
  description: string = "";
  category: string = "";
  city: string = "";
  state: string = "";
  price: number | null = null;
}

export interface ILiker {
  username: string;
  displayName: string;
  image: string;
  isChef: boolean;
}

export interface IPhoto {
  id: string;
  url: string;
  isMain: boolean;
}
